.evergreen/run-serverless-tests.sh

```diff
@@ -12,10 +12,10 @@ if [ -z ${SERVERLESS_ATLAS_USER+omitted} ]; then echo "SERVERLESS_ATLAS_USER is
 if [ -z ${SERVERLESS_ATLAS_PASSWORD+omitted} ]; then echo "SERVERLESS_ATLAS_PASSWORD is unset" && exit 1; fi
 
 npx mocha --file test/tools/runner/index.js \
```

Comment:
For every line that has been updated here, we need to make sure all the tests still run and pass as part of the `test-test-serverless` task in evergreen (reviewers: please compare the tests that run on this PR against the tests in waterfall, for this task, which runs under the "Serverless Test" configuration in evergreen)

Reply:
Even better - this PR has serverless running more tests than main. Main was missing retryable reads and writes it looks like.

Reply:
Looks like the same 400 in main and this PR, so good there.